sigridci/sigridci.py

```diff
@@ -105,17 +112,27 @@ def callSigridAPI(self, api, path):
             log("Received empty response")
             return {}
         return json.loads(responseBody)
-
-    def getTokenHeaderValue(self):
-        token = os.environ["SIGRID_CI_TOKEN"]
-        if len(token) >= 32:
-            return f"Bearer {token}".encode("utf8")
-        else:
-            log("WARNING: You are using an outdated Sigrid CI token that will be deactivated soon.")
-            log("         See https://github.com/Software-Improvement-Group/sigridci/blob/main/docs/authentication-tokens.md")
-            log("         for instructions on how you can create a new token.")
-            account = os.environ["SIGRID_CI_ACCOUNT"]
-            return b"Basic " + base64.standard_b64encode(f"{account}:{token}".encode("utf8"))
+        
+    def retry(self, operation, *, attempts=5, allow404=False, allowEmpty=True):
+        for attempt in range(attempts):
+            try:
+                response = operation()
+                if response != {} or allowEmpty:
+                    return response
+            except urllib.error.HTTPError as e:
+                if e.code in [401, 403]:
+                    log("You are not authorized to access Sigrid for this system")
+                    sys.exit(1)
+                elif e.code == 404 and allow404:
```

Comment:
To me it seems more logical to swap these:
```suggestion
                elif allow404 and e.code == 404:
```